js/core/src/Blocks.svelte

```diff
@@ -188,16 +190,26 @@
 		add_to_api_calls
 	);
 
-	let old_dependencies = dependencies;
-	// $: if (
-	// 	dependencies !== old_dependencies &&
-	// 	render_complete &&
-	// 	!layout_creating
-	// ) {
-	// 	// re-run load triggers in SSR mode when page changes
-	// 	// handle_load_triggers();
-	// 	old_dependencies = dependencies;
-	// }
+	$effect(() => {
+		reload_count;
```

Comment:
This is probably sacrilegious but let me explain myself.

My first thought was to make app_tree and dep_manager state variables. However, I couldn't quite get this to work. Some things that went wrong:
* the app_tree was created more than once
* dep_manager and app_tree would get out of synch
* I tried creating both in one $derived.by block but that caused problems because the setContext cant be called in a derived block and without it the dep manager can't get the state of a component.

So what I did instead is "manually" update the app_tree and dep_manager state when we're sure a reload should happen.